LDAR_Sim/testing/end_to_end_testing/e2e_suite_runner.py

```diff
@@ -38,214 +34,42 @@
-                    print(rm.SIM_SET.format(simulation_number=simulation_number))
-                    # read in pregen emissions
-                    infra = read_in_emissions(infrastructure, generator_dir, simulation_number)
-                    # -- Run simulations --
-                    prog_data = []
-                    for program in programs:
-                        meth_params = {}
-                        prog_measured_df = filter_deployment_tf_by_program_methods(
-                            site_measured, programs[program][pdc.Program_Params.METHODS]
-                        )
-                        for meth in programs[program][pdc.Program_Params.METHODS]:
-                            meth_params[meth] = methods[meth]
-                        prog_data.append(
-                            (
-                                daylight,
-                                weather,
-                                simulation_number,
-                                program,
-                                meth_params,
-                                programs[program],
-                                sim_params,
-                                virtual_world,
-                                output_params,
-                                infra,
-                                in_dir,
-                                out_dir,
-                                seed_timeseries,
-                                lock,
-                                prog_measured_df,
-                            )
-                        )
-
-                    with mp.Pool(processes=n_process) as p:
-                        _ = p.starmap(
-                            simulate,
-                            prog_data,
-                        )
-                    gc.collect()
-                    print(rm.FIN_SIM_SET.format(simulation_number=simulation_number))
-
-                # -- Batch Report --
-                print(rm.BATCH_CLEAN.format(batch_count=0))
-                summary_stats_manager.gen_summary_outputs(False)
-                non_baseline_progs = get_non_baseline_prog_names(programs, base_program)
-                summary_stats_manager.gen_cost_summary_outputs(non_baseline_progs)
-            # Compare results to expected
-            compare_outputs(test.name, out_dir, expected_results)
+
+            simulation_manager.check_inputs()
+
+            simulation_manager.initialize_outputs(input_manager)
+
+            simulation_manager.check_generator_files()
+
+            simulation_manager.setup_infrastructure()
+
+            simulation_manager.setup_emissions()
+
+            simulation_manager.setup_weather()
+
+            simulation_manager.setup_daylight()
+
+            simulation_manager.run_simulations(DEBUG=False)
+
+            simulation_manager.generate_summary_results()
+
+            simulation_manager.validate_outputs(test=test, comparison_function=compare_outputs)
```

Comment:
Is there a reason for wrapping this function in another function? With the current code, it just feels like unnecessary wrapping.

Reply:
Just to keep management of results the job of the manager instead of the the script for separation of responsibilities